### PR TITLE
Adding foreman-installer man page

### PR DESCRIPTION
**Pull request by lzap**
It builds fine

http://koji.katello.org/koji/taskinfo?taskID=66648
http://koji.katello.org/koji/taskinfo?taskID=66658

_but_ rubygem-logging in F19 is missing rubygem-multi_json dependency. Koji is
picking up rubygem-logging-1.8.1-22.fc19 which has missing dep. I need to look
on it tomorrow.

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
Any chance of adding the man page to the debs too? :)

**Comment by lzap** [comment]
Fixed wrong deps for F19, now it builds fine even without any hack in foreman-nightly-fedora19 http://koji.katello.org/koji/taskinfo?taskID=66832

Working on deb man page.

**Comment by domcleal** [comment]
Something I noticed in the finished man page was FIXMEs in the headers/footers, where do they need setting?

**Comment by lzap** [comment]
Yes, thanks, I just pushed a fix. Note that you need newest kafo, there are some issues in the man page with "Parameters for 'modulename'" - its unproperly rendered there. With the latest kafo (master) it is fine.

**Comment by lzap** [comment]
Ok pushing something new - I "rakified" this project, now we have _much_ cleaner SPEC file. I will work now to simplify the Debian one (@ares @GregSutcliffe some help over here)?

The SPEC file is now as simple as that:

```
%build
rake build \
  PREFIX=%{_prefix} \
  SBINDIR=%{_sbindir} \
  SYSCONFDIR=%{_sysconfdir} \
  --trace

%install
rake install \
  PREFIX=%{buildroot}%{_prefix} \
  SBINDIR=%{buildroot}%{_sbindir} \
  SYSCONFDIR=%{buildroot}%{_sysconfdir} \
  --trace
```

And yes, @abenari, it is now possible to compile/install this project using simple rake:

```
rake install
```

By default it installs to `/usr/local`.

I would like to also "rakefify" main app and proxy. Opinions?

**Comment by lzap** [comment]
Guys I am done, can you please merge this so I can proceed with Debian package? I have my PR locally now.

**Comment by ekohl** [comment]
Not an expert on Rakefiles and Ruby, but I think it looks good. It can remove a lot of duplication with the packaging and make porting to other distros a lot easier.

**Comment by lzap** [comment]
Yep for example for Debian I will remove ALL the required patches there, nice and clean.

This Rakefile is not a best example as we do some man page generation magic with kafo there, but the second part of the Rakefile looks much better. Almost a Makefile, I like that. I would like to know your opinions if you would like this also for proxy and core package.

**Comment by lzap** [comment]
@domcleal ping? Can you waive it so I can merge?

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+task :install => :build do |t|
+  mkdir_p "#{DATADIR}/foreman-installer"
+  cp_r Dir.glob('{checks,config,modules,VERSION,README.md,LICENSE}'), "#{DATADIR}/foreman-installer"
+
+  mkdir_p "#{SYSCONFDIR}/foreman"
+  cp "#{BUILDDIR}/foreman-installer.yaml", "#{SYSCONFDIR}/foreman/"
+  cp "config/answers.yaml", "#{SYSCONFDIR}/foreman/foreman-installer-answers.yaml"
+
+  mkdir_p SBINDIR
+  install "#{BUILDDIR}/foreman-installer", "#{SBINDIR}/foreman-installer", :mode => 0777, :verbose => true
```
Comment (domcleal) [inline Rakefile]: eeek. `-rwxrwxrwx   1 root root  3075 Dec  3 10:31 /usr/sbin/foreman-installer`

File: man/foreman-installer.8.asciidoc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+Run puppet in noop mode? (default: false)
+
+-v, --verbose::
+Display log on STDOUT instead of progressbar
+
+-l, --verbose-log-level 'LEVEL'::
+Log level for verbose mode output (default: "info")
+
+--ignore-undocumented::
+Ignore inconsistent parameters documentation (default: false)
```
Comment (domcleal) [inline man/foreman-installer.8.asciidoc]: "parameter documentation"

File: man/foreman-installer.8.asciidoc [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+Log level for log file output (default: :debug)
+
+--[no-]colors::
+Use color output on STDOUT (default: true)
+
+-h, --help::
+Print help
+
+--[no-]enable-foreman::
+Enable puppet module foreman? (default: true)
```
Comment (domcleal) [inline man/foreman-installer.8.asciidoc]: "Enable foreman puppet module".  Are you able to do something like `<code>` around the word "foreman", e.g. bold or backticks to make it clear this is the name of a module?

File: man/foreman-installer.8.asciidoc [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+--[no-]enable-foreman-proxy::
+Enable puppet module foreman_proxy? (default: true)
+
+--[no-]enable-puppet::
+Enable puppet module puppet? (default: true)
```
Comment (domcleal) [inline man/foreman-installer.8.asciidoc]: It's more of an issue here where we have a puppet puppet module..

File: man/foreman-installer.8.asciidoc [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+
+`Parameters for modules`
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following parameters are available as puppet parameters:
```
Comment (domcleal) [inline man/foreman-installer.8.asciidoc]: s/puppet parameters/installer arguments/ or installer parameters?

File: man/foreman-installer.8.asciidoc [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        --foreman-proxy-dhcp=true \
+        --foreman-proxy-dhcp-range="192.168.100.10 192.168.100.200" \
+        --foreman-proxy-dns=true \
+        --foreman-proxy-dns-forwarders=192.168.100.1
+
+FILES
+-----
+
+`/etc/foreman/foreman-installer.yaml`::
+Foreman-installer is based on 'kafo' project. Contains configuration for
```
Comment (domcleal) [inline man/foreman-installer.8.asciidoc]: the kafo

Comment (lzap) [inline man/foreman-installer.8.asciidoc]: All the above fixed (also upstream in kafo).

Give me a go, I can merge.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by domcleal** [comment]
Merged as 721a83a, thanks @lzap.

**Pull request closed without merging**